### ISSUE TITLE
Per page titles in html output

### ISSUE THREAD

**Issue by benwhalley** [rstudio/bookdown#544]
Related to issue #7, is there a way of having a per-page title in the html output?  Perhaps with the overall book title as a prefix, and the title attr of the individual page specified in the Rmd (or by section title)?

At present google results for individual pages are indistinguishable, even when there are nice chapter or section titles which could be used.

I also wonder if this could be linked to the way running headers are implemented for latex output?

**Comment by benwhalley** [comment]
Just to add, as requested:

```
Session info -------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.1 (2017-06-30)
 system   x86_64, darwin16.7.0        
 ui       RStudio (1.0.153)           
 language (EN)                        
 collate  en_GB.UTF-8                 
 tz       Europe/London               
 date     2018-03-08                  

Packages -----------------------------------------------------------------------------------
 package   * version date       source        
 backports   1.1.2   2017-12-13 CRAN (R 3.4.1)
 base64enc   0.1-3   2015-07-28 CRAN (R 3.4.1)
 bookdown    0.7     2018-02-18 CRAN (R 3.4.1)
 digest      0.6.15  2018-01-28 CRAN (R 3.4.1)
 evaluate    0.10.1  2017-06-24 CRAN (R 3.4.1)
 glue        1.2.0   2017-10-29 cran (@1.2.0) 
 graphics  * 3.4.1   2017-08-18 local         
 grDevices * 3.4.1   2017-08-18 local         
 highr       0.6     2016-05-09 CRAN (R 3.4.1)
 htmltools   0.3.6   2017-04-28 CRAN (R 3.4.1)
 jsonlite    1.5     2017-06-01 CRAN (R 3.4.1)
 knitr     * 1.20    2018-02-20 CRAN (R 3.4.1)
 magrittr    1.5     2014-11-22 CRAN (R 3.4.1)
 markdown    0.8     2017-04-20 CRAN (R 3.4.1)
 methods   * 3.4.1   2017-08-18 local         
 mime        0.5     2016-07-07 CRAN (R 3.4.1)
 Rcpp      * 0.12.15 2018-01-20 CRAN (R 3.4.1)
 rmarkdown   1.9     2018-03-01 CRAN (R 3.4.1)
 rprojroot   1.3-2   2018-01-03 CRAN (R 3.4.1)
 stats     * 3.4.1   2017-08-18 local         
 stringi   * 1.1.6   2017-11-17 CRAN (R 3.4.1)
 stringr   * 1.3.0   2018-02-19 CRAN (R 3.4.1)
 tinytex     0.3     2018-01-22 CRAN (R 3.4.1)
 tools     * 3.4.1   2017-08-18 local         
 utils     * 3.4.1   2017-08-18 local         
 xfun        0.1     2018-01-22 CRAN (R 3.4.1)
 yaml        2.1.17  2018-02-27 CRAN (R 3.4.1)
[1] ‘1.19.2.1’
pdfTeX 3.14159265-2.6-1.40.18 (TeX Live 2017)
kpathsea version 6.2.3
Copyright 2017 Han The Thanh (pdfTeX) et al.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the pdfTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the pdfTeX source.
Primary author of pdfTeX: Han The Thanh (pdfTeX) et al.
Compiled with libpng 1.6.29; using libpng 1.6.29
Compiled with zlib 1.2.11; using zlib 1.2.11
Compiled with xpdf version 3.04

```

**Comment by yihui** [comment]
I agree this will be a very useful feature, but I don't think I'll have time to implement it until after this summer. Pull requests are welcome! :)

**Comment by batpigandme** [comment]
*TL;DR* Proposing adding an "option" to use *article* as the `og:type`. If this seems like a decent idea to you, Yihui, I'd be willing to work on it for a PR. I _think_ this would allow me to use existing title and description properties and cat the chapter title, etc. for the metadata.

Related if not the same issue (I believe they're the same). I've been thinking about how to add per "chapter" metadata (specifically `og:*`), since currently all links to a book show up as the overall book.

From what I've [read](http://ogp.me/#types), this doesn't seem like an option with `og:type` as `book` which only has:
`book` - Namespace URI: http://ogp.me/ns/book#
* `book:author` - profile array - Who wrote this book.
* `book:isbn` - string - The ISBN
* `book:release_date` - datetime - The date the book was released.
* `book:tag` - string array - Tag words associated with this book.

Though it's an inaccurate descriptor in some sense, the `article` type (properties listed below) could have some advantages, especially for books under development, _e.g._ the ability to have `modified_time` would be helpful for knowing whether or not to revisit material if you've looked at the book before.

`article` - Namespace URI: http://ogp.me/ns/article#
* `article:published_time` - datetime - When the article was first published.
* `article:modified_time` - datetime - When the article was last changed.
* `article:expiration_time` - datetime - When the article is out of date after.
* `article:author` - profile array - Writers of the article.
* `article:section` - string - A high-level section name. E.g. Technology
* `article:tag` - string array - Tag words associated with this article.

Happy to open this as a separate issue, if you think that's appropriate.

Thanks!

**Comment by yihui** [comment]
@batpigandme I'm totally okay with `article`! I care much more about how a link looks on social media or anywhere that supports the open graph protocol, than whether the type is "accurate". For example, in @benwhalley's request, I think it makes perfect sense to display different content for different chapters.

**Comment by batpigandme** [comment]
Awesome. I'm kind of learning as I go, so I don't know how long it will take me, but I'd be happy to start plugging away at this one. Now that I can see click-throughs with the community site, I'm more convinced than ever that this would make a difference in terms of getting people to use existing resources, once they've been pointed in the right direction. 😄

**Comment by batpigandme** [comment]
**Status update:**

So I've gotten kind of stuck here – the opengraph info comes from the basic metadata for each page, and, in bookdown, the title attribute is the same for every chapter/page:
https://github.com/rstudio/bookdown/blob/52175179e63cebb1e2cc2931bada33a1c84b95e1/inst/templates/gitbook.html#L8
<img width="1691" alt="bookdown_title_attr" src="https://user-images.githubusercontent.com/831732/40500300-51ffae50-5f52-11e8-8786-cd4cc24ff2e3.png">

Going back to gitbook code, this doesn't seem to be the case (they don't have og, or twitter tags implemented, but I can see where I would pull the info from)
<img width="1760" alt="gitbook_title_attr" src="https://user-images.githubusercontent.com/831732/40500368-79affffe-5f52-11e8-8912-93c8df88cee7.png">

I tried to override this using an approach similar to that in pkgdown, but the title element keeps getting overwritten. @yihui, feel free to tell me to read an entire book — I really do want to understand how this works, I can just tell that I'm somehow coming at it from the completely wrong angle! 😬

**Comment by yihui** [comment]
@batpigandme If you don't mind, I can start working on this issue myself soon (next Tuesday maybe), since it bothers myself more and more (e.g., the card should have been more informative here https://community.rstudio.com/t/8441).

**Comment by batpigandme** [comment]
No problem! Thanks, @yihui!

**Comment by MateuszKalamarz** [comment]
@yihui @batpigandme I would like to ask - as the comment is in this issue - did you manage to provide an option to set **og:type** as **article** and to set other og and twitter parameters?
